src/server/publicApi.js

```diff
@@ -1,125 +1,181 @@
+            `${gameName}:${gameID}:metadata`
+        );
+        let model = await gameServer.db.get(`${gameName}:${gameID}:model`);
+
+        // update the model with the identified threats
+        Object.keys(res.G.identifiedThreats).forEach((diagramIdx) => {
+            Object.keys(res.G.identifiedThreats[diagramIdx]).forEach(
+                (componentIdx) => {
+                    let diagram = model.detail.diagrams[diagramIdx].diagramJson;
+                    let cell = null;
+                    for (let i = 0; i < diagram.cells.length; i++) {
+                        let c = diagram.cells[i];
+                        if (c.id === componentIdx) {
+                            cell = c;
+                            break;
+                        }
+                    }
+                    if (cell !== null) {
+                        let threats = [];
+                        if (Array.isArray(cell.threats)) {
+                            threats = cell.threats;
+                        }
+                        Object.keys(
+                            res.G.identifiedThreats[diagramIdx][componentIdx]
+                        ).forEach((threatIdx) => {
+                            let t =
+                                res.G.identifiedThreats[diagramIdx][componentIdx][threatIdx];
+                            threats.push({
+                                status: "Open",
+                                severity: t.severity,
+                                id: t.id,
+                                methodology: "STRIDE",
+                                type: getTypeString(t.type),
+                                title: t.title,
+                                description: t.description,
+                                mitigation: t.mitigation,
+                                owner: metadata.players[t.owner].name,
+                                game: gameID,
+                            });
+                        });
+                        cell.threats = threats;
+                    }
+                }
+            );
         });
+
+        const modelTitle = model.summary.title.replace(" ", "-");
+        const timestamp = new Date().toISOString().replace(":", "-");
+        ctx.attachment(`${modelTitle}-${timestamp}.json`);
+        ctx.body = model;
     });
 
-    const modelTitle = model.summary.title.replace(' ', '-');
-    const timestamp = (new Date()).toISOString().replace(":", "-");   
-    ctx.attachment(`${modelTitle}-${timestamp}.json`);
-    ctx.body = model;
+
+    //produce a nice textfile with the threats in
+    router.get("/download/text/:id", async (ctx) => {
+        //get some variables that might be useful
+        const gameName = ElevationOfPrivilege.name;
+        const gameID = ctx.params.id;
+        const res = await gameServer.db.get(`${gameName}:${gameID}`);
+        var threats = [];
+
+        Object.keys(res.G.identifiedThreats).forEach((diagramId) => {
```

Comment:
I believe we definitely need a test that this also works for complicated models.

Reply:
I improved the test and realised it didn't add the threats in the model so changed this